integration/tests/cook/test_basic.py

```diff
@@ -700,6 +700,41 @@ def test_detailed_for_groups(self):
         for k in detail_keys:
             self.assertIn(k, group_info)
 
+    def test_group_kill(self):
+        # Create and submit jobs in group
+        slow_job_wait = 1200
+        group_spec = self.minimal_group()
+        job_fast = util.minimal_job(group=group_spec['uuid'])
+        job_slow = util.minimal_job(group=group_spec['uuid'],
+                                    command='sleep %d' % slow_job_wait)
+        data = {'jobs': [job_fast, job_slow], 'groups': [group_spec]}
+        resp = util.session.post('%s/rawscheduler' % self.cook_url, json=data)
+        self.assertEqual(resp.status_code, 201)
+        self.logger.debug("group " + group_spec['uuid'])
+        self.logger.debug("fast  " + job_fast['uuid'])
+        self.logger.debug("slow  " + job_slow['uuid'])
+        # Wait for the short job to finish
+        util.wait_for_job(self.cook_url, job_fast['uuid'], 'completed')
+        # Now try to cancel the group (just the long job)
+        resp = util.session.delete('%s/group?uuid=%s' % (self.cook_url, group_spec['uuid']))
+        self.assertEqual(resp.status_code, 204)
+        # Wait for the slow job (and its instance) to die
+        util.wait_for_job(self.cook_url, job_slow['uuid'], 'completed')
+        util.wait_for_exit_code(self.cook_url, job_slow['uuid'])
+        # The fast job should have Success, slow job Failed (because we killed it)
+        jobs = util.session.get('%s/rawscheduler?job=%s&job=%s' %
+                                (self.cook_url, job_fast['uuid'], job_slow['uuid']))
+        self.assertEqual(200, jobs.status_code)
+        jobs = jobs.json()
+        self.logger.debug("JOBS:\n" + (json.dumps(jobs, sort_keys=True)))
+        self.assertEqual('success', jobs[0]['state'], 'Job details: %s' % (json.dumps(jobs[0], sort_keys=True)))
+        self.assertEqual('failed', jobs[1]['state'], 'Job details: %s' % (json.dumps(jobs[1], sort_keys=True)))
+        self.assertEqual(99003, jobs[1]['instances'][0]['reason_code'])
+        # Now try to kill the group again
+        # (ensure it still works when there are no live jobs)
+        resp = util.session.delete('%s/group?uuid=%s' % (self.cook_url, group_spec['uuid']))
+        self.assertEqual(resp.status_code, 204)
+
```

Comment:
Should we add a test that multiple jobs get cancelled? Maybe some waiting and some running?

Reply:
done